src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/DumpTests.cs

```diff
@@ -97,5 +101,77 @@ public Task DumpTest(DiagnosticPortConnectionMode mode, DumpType type)
                     runner.ConfigurationFromEnvironment.SetDumpTempFolder(dumpTempFolder);
                 });
         }
+
+        [ConditionalTheory(typeof(TestConditions), nameof(TestConditions.IsDumpSupported))]
+        [InlineData(DiagnosticPortConnectionMode.Connect, DumpType.Mini)]
+        public Task DiagSessionTest(DiagnosticPortConnectionMode mode, DumpType type)
+        {
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                mode,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (runner, client) =>
+                {
+                    int processId = await runner.ProcessIdTask;
+
+                    PackageMode packageMode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? PackageMode.DiagSession : PackageMode.IncludeDacDbi;
+
+                    using ResponseStreamHolder holder = await client.CaptureDumpAsync(processId, type, packageMode);
+                    Assert.NotNull(holder);
+
+                    // The dump operation may still be in progress but the process should still be discoverable.
+                    // If this check fails, then the dump operation is causing dotnet-monitor to not be able
+                    // to observe the process any more.
+                    ProcessInfo processInfo = await client.GetProcessAsync(processId);
+                    Assert.NotNull(processInfo);
+
+                    string tempDiagSessionPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+
+                    using var fileStream = new FileStream(tempDiagSessionPath, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.DeleteOnClose);
+                    await holder.Stream.CopyToAsync(fileStream);
+                    fileStream.Position = 0L;
+
+                    //Validate here
+                    using Package package = Package.Open(fileStream, FileMode.Open, FileAccess.Read);
+
+                    //2-4 parts:
+                    //Metadatata
+                    //dmp
+                    //[Linux only]
+                    //dac
+                    //dbi
+
+                    int expectedParts = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? 2 : 4;
+
+                    PackagePartCollection parts = package.GetParts();
+                    Assert.Equal(expectedParts, parts.Count());
+
+                    PackagePart metadata = parts.FirstOrDefault(p => p.Uri.ToString().Contains("metadata.xml"));
+                    Assert.NotNull(metadata);
+                    PackagePart dump = parts.FirstOrDefault(p => p.Uri.ToString().EndsWith(".dmp"));
+                    Assert.NotNull(dump);
```

Comment:
Won't this check fail on non-Windows systems since it produces a core dump without an extension?

Reply:
We scrape the intermediate file, and this change forces all intermediate files to have a dmp extension.